### PR TITLE
implementation: parser in rust

### PR DESCRIPTION
**Pull request by mobuchowski**
Description in proposal doc: https://github.com/OpenLineage/OpenLineage/pull/627

Signed-off-by: Maciej Obuchowski <obuchowski.maciej@gmail.com>

Closes: #582

### CODE REVIEW

File: .circleci/workflows/openlineage-java.yml [hunk 1]
```diff
@@ -9,11 +9,11 @@ workflows:
       - build-integration-spark:
           matrix:
             parameters:
-              spark-version: [ '2.4.6', '3.1.2', '3.2.1' ]
```
Comment (pawel-big-lebowski) [inline .circleci/workflows/openlineage-java.yml:12]: Looks like a wrong git conflict resolution.
Any particular reason for that?

File: integration/common/openlineage/common/sql/parser.py [hunk 2]
```diff
@@ -7,11 +7,81 @@
...
-from openlineage.common.models import DbTableName
 
 log = logging.getLogger(__name__)
 
 
+def provider():
+    return "python"
+
+
+class DbTableMeta:
```
Comment (pawel-big-lebowski) [inline integration/common/openlineage/common/sql/parser.py:18]: Putting this class into separate PR would really improve the readability of this one.

File: integration/sql/test_python.sh [hunk 3]
```diff
@@ -0,0 +1,11 @@
...
+
+pip install maturin
+maturin build
+pip install target/wheels/*.whl
+pip install pytest
+
+pytest -vv tests/python
+
+
+#docker run --rm -it --platform linux/arm64/v8 --entrypoint bash -v $PWD:/code $(docker build -q .)
```
Comment (pawel-big-lebowski) [inline integration/sql/test_python.sh:11]: is this OK?

File: integration/sql/tests/tests_tpcds.rs [hunk 4]
```diff
@@ -0,0 +1,790 @@
+use openlineage_sql::SqlMeta;
+
+#[macro_use]
+mod test_utils;
+use test_utils::*;
+
+#[test]
+fn test_tpcds_cte_query() {
```
Comment (pawel-big-lebowski) [inline integration/sql/tests/tests_tpcds.rs:8]: Those TPCDS tests are awesome 👍

**Review by pawel-big-lebowski — COMMENTED** [review]
Complex tpcds really prove the value of this approach. 
This looks great to me!
Some minor comments added.

File: .circleci/workflows/openlineage-java.yml (hunk 1 above)
Comment (mobuchowski) [reply .circleci/workflows/openlineage-java.yml]: Yes, my mistake in manual resolution :)

Fixed.

**Pull request closed without merging**